addons/l10n_mx/models/account_move.py

```diff
@@ -0,0 +1,84 @@
+# coding: utf-8
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, _
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    def _move_autocomplete_invoice_lines_values(self):
+        self._l10n_mx_get_import_tax_base_amount_lines()
+        res = super(
+            AccountMove, self)._move_autocomplete_invoice_lines_values()
+        return res
+
+    def _l10n_mx_get_import_tax_base_amount_lines(self):
+        """Appends two lines for each one of the lines where the quantity is
+        zero, as the value of the lines added are need in cash basis taxes to
+        properly compute the taxes paid"""
+        self.ensure_one()
+        foreign = self.line_ids.mapped('l10n_mx_invoice_broker_id.partner_id')
+        self.line_ids.filtered(
+            lambda line: line.partner_id in foreign).unlink()
+        aml_obj = self.env['account.move.line'].with_context(
+            check_move_validity=False, recompute=False)
+
+        lines_2_create = []
+        for line in self.line_ids.filtered(
+                lambda l: not l.quantity and l.tax_ids and
+                l.l10n_mx_invoice_broker_id):
+            if line.l10n_mx_invoice_broker_id == self:
+                self.message_post(body=_(
+                    'The Overseas Invoice cannot be the same that the'
+                    'related invoice'))
+                continue
+            tax_ids = self.env['account.tax']
```

Comment:
What about using the following way?
`tax_ids = line.tax_ids | line.tax_ids.mapped('children_tax_ids').filtered(lambda child: child.type_tax_use != 'none')`